monitor/localOperations.go

```diff
@@ -48,22 +48,21 @@ func GetAllLocal(rootDir string) (map[string]Monitor, mapset.Set, error) {
 	monitorsSet := mapset.NewSet()
 	monitorsMap := make(map[string]Monitor)
 	for _, file := range files {
-		var allLocalMonitors []Monitor
+		var monitor Monitor
```

Comment:
This will fail for users who has multiple monitors in a single YAML file right ?

Reply:
I believe so, this would be a breaking change.

Reply:
I am just afraid of this change, do we really need this ? Also, this will make users use case. There are users who might categorize monitors and then reduce files. If there are too many monitors it would make hard to manage 100s of YML files.

Reply:
So personally, I think 100s of yaml files is better than one huge yaml file. Most of our monitors are between 80-120 LOC. If we had 100s, that would equate to a yaml file of about 10k lines.

Lots of yaml backed programs like kubernetes usually use individual yaml files to configure their applications.

I'm not certain what you mean by this:
>There are users who might categorize monitors and then reduce files.

Reply:
> There are users who might categorize monitors and then reduce files.

Basically, let's say you've 100 monitors and but one would categorized file as product or feature and have multiple monitors in same file. As I mentioned earlier,`sync` is to just a utility to get started from existing monitors and then you can still choose to have all monitors in different files. 

IMO, I would rather let user choose how they would want to organize these monitors instead of not prefer to enforce this by default but?

Thoughts ?

Reply:
I think most users will want to sync before making changes since people might have made a change in the UI and the monitors yaml will be out of date. If you reorganize the output of the monitors from sync, then sync will not update those monitors. That's our workflow, we sync nightly and before making any changes to make sure all changes from the UI get populated.

Reply:
I think enforcing one's workflow to everyone is not great way. I think it would be best for users to come up with their own usecase and workflow.

In order to solve the use case, how about we change sync in a way that it goes through all YMLs and then if found it will update the same object and if not it will in last dump all new monitors which are being synced for the first time to the new file and then one can organize and later it the state will be persistated. This way it user can decided and sync correctly.

In a nutshell, following
* first time sync , dump everything into new file 
* User organize their monitors
* user modify / updates
* User sync again, in this scenario
 * verify first if this monitor exists, if it does ensure we update the same file
 * Find a delta and leftovers (i.e new monitors) will be dumped to new file and then one can organize in their comfortable use case.

My idea here is to not user to enforce a way rather just mechanism.

Reply:
I like your sync change and I'm on board.

One question: are you still opposed to keeping a yaml file per monitor? To solve the category organization problem users can organize the monitors by folder.

Reply:
I am not opposed, I feel it shouldn't be enforced by tool.  I would let users choose their flow, they can categorize per file or per features up-to them.